### PR TITLE
Add function that checks if function id represents a clinit.

### PR DESCRIPTION
**Pull request by jeannielynnmoulton**
A utility for checking whether the supplied function id corresponds to a
clinit.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by thomasspriggs — APPROVED** [review]
Unit test, as well please. You know it makes sense :)

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7f1830eb4f1cc2f96fd1c210c58cedfe7d2c3b12).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113106427

File: jbmc/src/java_bytecode/java_static_initializers.cpp [hunk 1]
```diff
@@ -73,6 +73,14 @@ bool is_clinit_wrapper_function(const irep_idt &function_id)
   return has_suffix(id2string(function_id), clinit_wrapper_suffix);
 }
 
+/// Check if function_id is a clinit
+/// \param function_id: some function identifier
+/// \return true if the passed identifier is a clinit
+bool is_clinit_function(const irep_idt &function_id)
```
Comment (LAJW) [inline jbmc/src/java_bytecode/java_static_initializers.cpp]: 🚫 Function not used.

**Review by LAJW — APPROVED** [review]

**Review by smowton — CHANGES_REQUESTED** [review]
Indeed, no unused code in the cbmc codebase please :)

**Comment by jeannielynnmoulton** [comment]
@smowton No problem, added.

**Comment by smowton** [comment]
Add to https://github.com/diffblue/cbmc/blob/develop/jbmc/unit/Makefile

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c227ab5ca65bb56d2099e5b0214fed9d9b1e3af5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113421780

**Review by smowton — APPROVED** [review]

**Pull request merged**